### DISCUSSION TITLE
Plausibility of LAN-based Emporia Vue2 with esphome firmware?

### DISCUSSION THREAD

**Discussion by catatung** [emporia-vue-local/esphome#167] [General]
Hi,

I've been setting up an Emporia Vue2 with the esphome firmware here.  I'm happy with it so far, however I try to avoid maintaining wifi whenever possible, be it for security and/or reliability.  I like things to be wired.  Plus, things are never really "wireless" because you need power, so I always prefer wired with POE.

As I understand it, the Emporia VUE2 box gathers data and ultimately ends up on the i2c bus to pins IO21 and IO22 on the ESP32 inside the unit.  Does anyone know what the plausibility would be for breaking out those i2c lines to a separate ESP32, such as the WT32-ETH01 unit which works with LAN, and loading this firmware onto it to interpret and send the info via API or MQTT as we are doing here (but with wifi)?

I would like to run this as wired connectivity if possible, and willing to do some experimentation in this regard, as long as there isn't reason why it wouldn't work.  I know lots of you are very close to this unit and knows well about how it works from all the reverse engineering efforts that went on.

Thanks for any advice/info!
Randy

**Comment by ozzy** [comment]
hI @catatung I am new to this myself and have just flashed a rev5 without problems.  An easier approach I think would for you to tap into the serial output in stead of i2c as the data has already been decoded.

Do you currently have mqtt working as I have a problem ?

Best regards oz

**Reply by catatung** [reply]
Hi Ozzy,

I am familiar with MQTT in general however I previously flashed this to use the API between the emporia and home assistant esphome add-on and so I have not seen esphome/emporiavue with this custom firmware operate with MQTT yet.  I am looking to switch to MQTT eventually though to see if there is any appreciable different in performance or compatibility.  I would like to compare.

Randy

**Reply by ozzy** [reply]
Hi @catatung ,

Just got MQTT working see https://github.com/emporia-vue-local/esphome/discussions/166
I failed to open MQTT port on ufw and mosquitto on my system requires a config rule if you need to connect to the ip4 address.

Oz

**Comment by flaviut** [comment]
Don't see why that wouldn't work. But just to make sure you're aware,
electrical code in the USA would prohibit this because you're mixing low
and high voltage on a single box.

**Reply by ozzy** [reply]
Same in uk but an opto-isolator would be required.

P.S. Sorted my MQTT problem.

**Reply by catatung** [reply]
Hi Flaviut;

This is interesting to hear, however in my mind I had thoughts of wiring an external connector in place of the wifi antenna which would contain the i2c data lines and perhaps a gpio ground, should it be necessary for an external esp32 to read the i2c correctly.  The external ESP32 would then be in it's own enclosure and plug into this connector on the esp32 box in order to receive the data on the i2c lines.

As I understand it, there are two i2c lines within the emporia box;  Do you know does the ESP32/custom firmware require both i2c lines in order to receive sufficient data to operate?  Also, do you know if the firmware is hard coded to access the wifi interface, or does it use a library that utilizes whatever TCPIP connection it sees available?  Just wondering if the firmware would need to be modified in order to make use of the LAN connection rather than the wifi.  That bit would be a little out of my wheelhouse.

Thanks for your input
Randy

**Comment by deggle** [comment]
Just posting here to say I'm looking at the same but found the other post first so replied to that:
https://github.com/emporia-vue-local/esphome/discussions/67

Has anyone got any further with ethernet?
